src/api/query-helpers.ts

```diff
@@ -201,3 +202,20 @@ export function parseTraitAbi(req: Request, res: Response, next: NextFunction):
   }
   handleBadRequest(res, next, `Invalid 'trait_abi'`);
 }
+
+export function validateRequestHexInput(hash: string) {
+  try {
+    hexToBuffer(hash);
```

Comment:
The `hexToBuffer` function only does basic checking for input validation (the 0x prefix, even number of chars), but it doesn't account for situations like `Buffer.from('boop', 'hex').toString('hex') === ''` (invalid hex chars are generally ignored without throwing). 

I think we should perform that check within this `validateRequestHexInput` function since it will add a bit of overhead, and lots of other code calls `hexToBuffer` with known good data.

There's a few ways to validate if the `hash` string is valid hex chars. I'd run a quick benchmark (or find a resource if someone else has) to pick a faster one. For example: regex, `Buffer.from(hash, 'hex').toString('hex') === hash).toLowerCase()`, etc.